### ISSUE TITLE
macOS tkinter clipboard only works once

### ISSUE THREAD

**Issue by Goldthing** [python/cpython#104613]
When attempting to copy to the clipboard on macOS I can get it to work once and only once. 

Here's a bare miniumum example to illustrate the issue

```
import tkinter as tk
import datetime

def on_click():
    root.clipboard_clear()
    now = datetime.datetime.now()
    print(now)
    root.clipboard_append(now)

root = tk.Tk()
frame = tk.Frame(root, padx=10, pady=10)
frame.pack()
btn = tk.Button(frame, text="Copy time", command=on_click)
btn.pack()
root.mainloop()
```
The first time the button is pressed it prints the correct time and can also paste the results elsewhere e.g. a text editor. Any subsequent press of the button prints a new time, but attempting to paste elsewhere simply results in the first time being printed, the clipboard never updates. Closing and restarting the example resets the problem.

This appears to be a macOS specific bug as I have tested on windows 10 and Linux mint in a virtual box, both work as expected with me able to paste a new time on each subsequent button press. On Mac I have tested Python 3.10 & 3.11 on both x86 and M1 (running different OS versions) installed via downloading the installer and via homebrew. All combinations tried so far give me the same issue so hopefully this should be easy to reproduce.

**Comment by ned-deily** [comment]
One additional thought: how are you running your tests?  From a terminal shell? Or from IDLE?

**Comment by Goldthing** [comment]
Thanks for looking into this. All my tests have been done in terminal. I'm not on macOS 13 yet, so that's something I may have to explore, though my old x86 mac won't be able to get that far, I think macOS 11 is it's max. Will also look into my Tk versions.
I'll report back hopefully tomorrow when I get a chance to dig into it further and give full details of the systems/configurations it doesn't seem to work for. Very strange that I have 2 different architectures with different OS versions that are doing the same thing, maybe I have something installed that's causing it.

**Comment by ned-deily** [comment]
FWIW, your test also seems to work correctly with the current python.org 3.11.3 on a macOS 11.7.6 Intel system.

**Comment by Goldthing** [comment]
@ned-deily thank you again for taking the time to look into this, and the fact that you didn't reproduce my results gave me reason to completly reassess what I had and hadn't eliminated as common factors.

 I now think I have found my common factor, and it's not what I expected at all. It would appear that this ONLY occurs when pasting clipboard contents from Python/Tkinter into BBedit. I've never done anything in Tcl, but I may try to see if I can get a bare minimum together to further narrow down the problem.

I've been using BBedit for years and it's proven to be an otherwise very stable workhorse so somehow I just didn't think to paste elsewhere. What's truly bizzare is that I haven't been able to reproduce this in any other manner, every other copy/paste to BBedit works exactly as expected. For example I have now tried a similar operation to my initial bare minimum in terminal using `date +%s | pbcopy` to repeatedly load new text to the clipboard with no issues. Similarly copy/pasting from python to any other software I've now tried such as textedit, sublime, libre office and notes all work fine.

My next step is to pass everything on to the support team at BBedit to see if they can figure out hows this happens. It's still possible that the bug does sit somewhere on the Python/Tkinter or Tcl side, but I think their dev team are best placed to investigate any further as this is clearly a very niche bug.

**Comment by ned-deily** [comment]
That's certainly an unexpected finding! I agree that pursuing the problem with the BBEdit team is the best next step. Since I think it highly unlikely that this will turn out to be a Python issue, I'm going to close this one now but feel free to reopen it if it turns out otherwise.

**Comment by ned-deily** [comment]
Thanks for the analysis. The reason you can't find the code in Python itself is that tkinter is basically a thin glue layer to Tcl and Tk API calls and these clipboard functions are implemented within the Tcl and Tk libraries. So if someone wants to pursue this, the next step would be to produce a pure Tcl test case that demonstrates the problem and, if it also fails, bring it up on the Tcl issue tracker.

**Comment by siegel** [comment]
It appears that Tk's pasteboard support has had some [issues in the past](https://core.tcl-lang.org/tk/tktview/b947864419c55e7278b8) — no shade on them, it's hard to get this stuff right. :-)

At this point the underlying issue is probably well beyond any of our respective pay grades to chase down; it might be worth the OP filing a ticket against Tk and having the Tk team debug it from their side.

**Comment by Goldthing** [comment]
Thank you both, @siegel I will gladly follow this issue on to the Tk team, but you are right this is well beyond my non-existent pay grade :-)
Hopefully someone on their team will have some insight, though at this point we are so far beyond my comfort zone I'm not sure how much help I can be beyond raising the issue.

**Comment by Goldthing** [comment]
As an update for @ned-deily or anyone else coming to this thread I have now raised a [ticket](https://core.tcl-lang.org/tk/tktview/9990770674adf8ffb6bb89d4b9dcfda14ae55697) over with Tk. 

This has already been reproduced independantley using the base Tk text widget demo and [Pasteboard-Viewer](https://github.com/sindresorhus/Pasteboard-Viewer). Hopefully this means a fix will be found and merged into a future release.
